assignment4/Q1.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <vector>
+#include <set>
+
+using namespace std;
+//iterating two times - first to copy the board
+//not sure of its correctness
+//probably something is missing but I do not know yet
+
+struct Point {
+    int x;
+    int y;
+    char val;
+    int set_no;
+};
+
+//returns the set whom point belongs to
+int findSet(struct Point point) {
+    return point.set_no;
+}
```

Comment:
I agree, I thought that I will add something more to it and sadly did not have enough time for changing the code.